pulsar/producer.go

```diff
@@ -76,6 +76,15 @@ type ProducerOptions struct {
 	// This properties will be visible in the topic stats
 	Properties map[string]string
 
+	// SendTimeout set the timeout for a message that not be acknowledged by server since sent.
+	// Send and SendAsync returns an error after timeout.
+	// Default is 30 seconds, -1 to disable.
+	SendTimeout time.Duration
+
+	// BlockIfQueueFull control whether Send and SendAsync return error if producer's message queue is full.
+	// Default is false.
+	BlockIfQueueFull bool
```

Comment:
The logic should be negated here. The default should be to block if queue full.

Reply:
Thanks for review.

I intented to maintain the same default behavior as [client-java](https://pulsar.apache.org/docs/en/client-libraries-java/#configure-producer), but this does change the default behavior of current client-go.

How about renaming `BlockIfQueueFull` to `NonBlockIfQueueFull`? just like `DisableBatching` for enable batch by default.

I think the reason for doing this is, we couldn't check whether the user sets `false` explicitly or default `false`